### PR TITLE
Domains: Add plan upsell that explains new domains cannot be primary

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR adds a plan upsell banner that explains new domains on a free site will redirect to the free subdomain.

#### Testing instructions

For a site without a plan, go to Manage -> Domains, and then click on the 'Add a domain to this site' button.

You should see this banner above the results:

Screenshot 2020-05-07 at 9 09 45 PM

Verify that clicking the Upgrade button sends you to the checkout page with a Personal plan added to cart.

Also check that events are sent properly

Finally, make sure this banner won't show for sites with a plan, or for users that are exempt from PWPO

### CODE REVIEW

**Comment by fditrapani** [comment]
Thanks for putting this together! I confirmed that the banner showed up on the search and it took me right to the cart with the personal plan. 

I also confirmed that the following events fired which will be perfect:
- `calypso_new_domain_will_redirect_notice_upsell_impression`
- `calypso_new_domain_will_redirect_notice_upsell_upgrade_click`)

File: client/components/banner/index.jsx [hunk 1]
```diff
@@ -223,15 +225,15 @@ export class Banner extends Component {
 						) }
 						{ callToAction &&
 							( forceHref ? (
-								<Button compact primary target={ target }>
+								<Button compact primary={ this.props.primaryButton } target={ target }>
```
Comment (hambai) [inline client/components/banner/index.jsx:228]: More of a styling suggestion but you could destruct `primaryButton` above with the rest of the props

File: client/my-sites/domains/domain-management/components/domain/new-domains-redirection-notice-upsell.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			! selectedSite ||
+			isOnPaidPlan ||
+			! hasNonPrimaryDomainsFlag ||
+			hasCustomPrimaryDomain
+		) {
+			return null;
+		}
+
+		return (
+			<Banner
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/components/domain/new-domains-redirection-notice-upsell.jsx:47]: I'd guess this should be the `UpsellNudge` component instead - seee https://github.com/Automattic/wp-calypso/pull/41892 and https://github.com/Automattic/wp-calypso/issues/38778 for related work (cc @sixhours)

File: client/my-sites/domains/domain-search/index.jsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ import { getSuggestionsVendor } from 'lib/domains/suggestions';
  * Style dependencies
  */
 import './style.scss';
+import NewDomainsRedirectionNoticeUpsell from '../domain-management/components/domain/new-domains-redirection-notice-upsell';
```
Comment (hambai) [inline client/my-sites/domains/domain-search/index.jsx:45]: Another tiny thing - tis should not be in the "Style dependencies" block. Also I wonder if relative imports are good/bad thing (though I acknowledge I did something similar in the domain status code)

Comment (eltongo) [reply client/my-sites/domains/domain-search/index.jsx]: I blame PHPStorm :oldmanyellsatsky:

File: client/my-sites/domains/domain-management/components/domain/new-domains-redirection-notice-upsell.jsx (hunk 2 above)
Comment (eltongo) [reply client/my-sites/domains/domain-management/components/domain/new-domains-redirection-notice-upsell.jsx]: Nice. Good thing the two components were pretty much interchangeable :)

**Comment by eltongo** [comment]
@sixhours I've added you as a reviewer as well, just in case you have any thoughts/concerns about my change to UpsellNudge

**Comment by sixhours** [comment]
Hey hey @eltongo! Thanks for the ping. The goal with unifying upsells with `UpsellNudge` is to standardize how upsells look across Calypso, so this new upsell should use the same icon and colors for easy visual recognition. I'm in the process of refactoring `Banner` and `UpsellNudge` so they each have a clearer purpose (see #41955). Let me know if you have any questions!

**Comment by eltongo** [comment]
I'm going to deploy these changes as is for now, and as the designs are consolidated, we'll revisit this banner with any changes necessary. Thanks!

**Pull request merged**